src/DicomWebClient/API/IWadoService.cs

```diff
@@ -0,0 +1,150 @@
+        Task<DicomFile> Retrieve(
+            string studyInstanceUid,
+            string seriesInstanceUid,
+            string sopInstanceUid,
+            params DicomTransferSyntax[] transferSyntaxes);
+
+        /// <summary>
+        /// Retrieves the metadata of the specified DICOM instance.
+        /// </summary>
+        /// <typeparam name="T">T must be type of <see cref="DicomDataset"/> or <see cref="string"/></typeparam>
+        /// <param name="studyInstanceUid">Study Instance UID</param>
+        /// <param name="seriesInstanceUid">Series Instance UID</param>
+        /// <param name="sopInstanceUid">SOP Instance UID</param>
+        /// <returns>A <c>T</c> containing DICOM metadata for the instance.</returns>
+        Task<T> RetrieveMetadata<T>(
+            string studyInstanceUid,
+            string seriesInstanceUid,
+            string sopInstanceUid);
+
+        /// <summary>
+        /// Retrieves one or more frames from a multi-frame DICOM instance.
+        /// </summary>
+        /// <param name="studyInstanceUid">Study Instance UID</param>
+        /// <param name="seriesInstanceUid">Series Instance UID</param>
+        /// <param name="sopInstanceUid">SOP Instance UID</param>
+        /// <param name="frameNumbers">The frames to retrieve within a multi-frame instance. (One-based indices)</param>
+        /// <param name="transferSyntaxes">An array of supported transfer syntaxes. Default set to Explicit VR Little Endian (1.2.840.10008.1.2.1)</param>
+        /// <returns>A <see cref="DicomFile"/> representing the DICOM instance.</returns>
+        Task<DicomFile> Retrieve(
+            string studyInstanceUid,
+            string seriesInstanceUid,
+            string sopInstanceUid,
+            uint[] frameNumbers,
+            params DicomTransferSyntax[] transferSyntaxes);
+
+        /// <summary>
+        /// Retrieves bulkdata in a DICOM instance.
+        /// </summary>
+        /// <param name="studyInstanceUid">Study Instance UID</param>
+        /// <param name="seriesInstanceUid">Series Instance UID</param>
+        /// <param name="sopInstanceUid">SOP Instance UID</param>
+        /// <param name="dicomTag">DICOM tag containing to bulkdata</param>
+        /// <param name="byteRange">Range of data to retrieve.
+        /// Entire range if <c>null</c>.
+        /// If <c>byteRange.Item2</c> is not specified then value specified in <c>byteRange.Item1</c>(start) to the end is retrieved.</param>
+        /// <param name="transferSyntaxes">An array of supported transfer syntaxes to be used to encode the bulkdata. Default set to Explicit VR Little Endian (1.2.840.10008.1.2.1)</param>
+        /// <returns>A byte array containing the bulkdata.</returns>
+        Task<byte[]> Retrieve(
+            string studyInstanceUid,
+            string seriesInstanceUid,
+            string sopInstanceUid,
+            DicomTag dicomTag,
+            Tuple<int, int?> byteRange = null,
```

Comment:
nit: be aware that default parameters are assigned at compilation, so if the signature changes callers will magically break with a runtime exception that doesn't make much sense.